packages/lib/src/vis/line/DataCurve.tsx

```diff
@@ -52,13 +57,26 @@ function DataCurve(props: Props) {
     invalidate();
   }, [dataGeometry, invalidate, points.data]);
 
+  useRaycasterThreshold('Line', LINE_WIDTH);
+
+  const handleClick = useCallback(
+    (evt: ThreeEvent<MouseEvent>) => {
+      const { index } = evt;
+
+      if (onLineClick && index !== undefined) {
+        onLineClick(index, evt);
+      }
+    },
+    [onLineClick]
+  );
+
   const showLine = visible && curveType !== CurveType.GlyphsOnly;
   const showGlyphs = visible && curveType !== CurveType.LineOnly;
 
   return (
     <>
-      <line_ visible={showLine} geometry={dataGeometry}>
-        <lineBasicMaterial color={color} linewidth={2} />
+      <line_ visible={showLine} geometry={dataGeometry} onClick={handleClick}>
```

Comment:
Just so the handler does not get registered if there's no callback:

```suggestion
      <line_ visible={showLine} geometry={dataGeometry} onClick={onLineClick && handleClick}>
```